[GXA-424] Add gene2experiment-populator Job for SolrCloud in SCXA

ke4 opened the pull request:
Introduce a Kubernetes Job to populate gene2experiment data into SolrCloud. Updates include new templates, ConfigMap enhancements, and values for controlling the job and collection configurations.

ke4 committed:
Add gene2experiment-populator Job for SolrCloud in SCXA

Introduce a Kubernetes Job to populate gene2experiment data into SolrCloud. Updates include new templates, ConfigMap enhancements, and values for controlling the job and collection configurations.

sandsebi commented on charts/scxa/templates/solrcloud-scxa-gene2experiment-populator.yaml:
```diff
@@ -0,0 +1,121 @@
+              valueFrom:
+                configMapKeyRef:
+                  name: {{ include "app.fullname" . }}-config
+                  key: SOLR_HOST
```
Do you need both SOLR_HOST and SOLR_HOSTS ?

ke4 replied:
Yes, unfortunately the scripts inside `index-scxa` repo are using both.

sandsebi commented on charts/scxa/templates/solrcloud-scxa-gene2experiment-populator.yaml:
```diff
@@ -0,0 +1,121 @@
+{{- if and .Values.solr .Values.solr.gene2experiment.populator .Values.solr.scxaAnalytics.populator.enabled | default false -}}
```
Is the precedence correct here? Or did you mean: if ((gene2experiment.populator) && (scxaAnalytics.populator.enabled)) | dafault false ? 
Because currently the | applies to just scxaAnalytics.populator.enabled if I am not mistaken.

ke4 replied:
@sandsebi Good catch! Thanks! I corrected this typo. ✅

sandsebi reviewed (approved):
A few suggestions, but otherwise it looks good 👍

ke4 committed:
Fix conditional check for gene2experiment populator job

Corrected the conditional logic to use the appropriate `enabled` field for the gene2experiment populator. This ensures the job is created only when explicitly enabled in the configuration.
